FlowCal/mef.py

```diff
@@ -308,38 +370,44 @@ def sc_fun(p,x):
             beads_model_str,
             beads_params_names)
 
-def plot_standard_curve(fl_channel,
+def plot_standard_curve(fl_au,
                         fl_mef,
                         beads_model,
                         std_crv,
+                        xscale='linear',
+                        yscale='linear',
                         xlim=None,
                         ylim=(1.,1e8)):
     """
     Plot a standard curve with fluorescence of calibration beads.
 
     Parameters
     ----------
-    fl_channel : array_like
-        Fluorescence of the calibration beads' subpopulations, in channel
-        numbers.
+    fl_au : array_like
+        Fluorescence of the calibration beads' subpopulations, in arbitrary
+        units.
     fl_mef : array_like
         Fluorescence of the calibration beads' subpopulations, in MEF
         units.
     beads_model : function
         Fluorescence model of the calibration beads.
     std_crv : function
-        The standard curve, mapping channel units to MEF units).
+        The standard curve, mapping arbitrary units to MEF units.
 
     Other parameters:
     -----------------
+    xscale : str, optional
+        Scale of the x axis, either ``linear`` or ``log``.
+    yscale : str, optional
+        Scale of the y axis, either ``linear`` or ``log``.
```

Comment:
Made a quick attempt at this, and these are some resulting plots:

Integer data:
![std_crv_fl1_b0002](https://cloud.githubusercontent.com/assets/1529481/16087287/1d7847fc-32e8-11e6-94fe-745710d5b217.png)

Floating-point data
![std_crv_apc-a_b3](https://cloud.githubusercontent.com/assets/1529481/16088187/911440dc-32eb-11e6-8ca8-73431ff7d559.png)

When I look at the standard curves, I try to quickly identify how many decades in a.u. are mapped to how many decades in MEF. These plots don't help me do that. In fact, I think the shape distortion associated with the logicle scale makes a lot of information not easily observable.

Another argument for keeping this plot in log scale is that the standard curve is a line in log scale, so this line should be easier to see if the plot itself is in log scale.

Finally, this function is not supposed to be accessed even by an API user, and instead used by `get_transform_fxn`. It is also not in the plot module, where I would agree it is a good idea to be consistent and make all functions work with logicle.

For all these reasons, I don't think it is a good idea to make this plot in logicle.